### DISCUSSION TITLE
Enable -fwarn-incomplete-patterns by default

### DISCUSSION THREAD

**Discussion by anka-213** [ghc-proposals/ghc-proposals#464] [Ideas]
Accidental partial pattern matching is a very common stumbling block for beginners and seasoned users alike. I think enabling the warning `-fwarn-incomplete-patterns` by default would mean less confusion for everyone.

For example, stackoverflow has over 200 questions from users that are confused by the runtime error:
https://stackoverflow.com/search?q=%22Non-exhaustive+patterns%22+%5Bhaskell%5D

I first created this as an issue on the ghc tracker, but people said it needs a proposal first:
https://gitlab.haskell.org/ghc/ghc/-/issues/18880

**Comment by goldfirere** [comment]
I would warmly welcome such a proposal. Go for it!
